### PR TITLE
8302791: Add specific ClassLoader object to Proxy IllegalArgumentException message

### PR DESCRIPTION
**Pull request by yathamravali**
Added specific class loader object to proxy IllegalArgumentException from which the class was not visible

The bug report for the same: https://bugs.openjdk.org/browse/JDK-8302791

OpenJDK PR: https://github.com/openjdk/jdk/pull/12641

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8302791](https://bugs.openjdk.org/browse/JDK-8302791): Add specific ClassLoader object to Proxy IllegalArgumentException message

### Reviewers
 * [Paul Hohensee](https://openjdk.org/census#phh) (@phohensee - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u-dev pull/290/head:pull/290` \
`$ git checkout pull/290`

Update a local copy of the PR: \
`$ git checkout pull/290` \
`$ git pull https://git.openjdk.org/jdk8u-dev pull/290/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 290`

View PR using the GUI difftool: \
`$ git pr show -t 290`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u-dev/pull/290.diff">https://git.openjdk.org/jdk8u-dev/pull/290.diff</a>

</details>

### CODE REVIEW

**Comment by yathamravali** [comment]
@mlchung @jerboaa - As in JDK 8 there is no getName() and nameAndId() functionalities, it would be helpful to have the ClassLoader string representation in the IllegalArgumentException message.

Please kindly review this PR, Thanks!

**Review by phohensee — APPROVED** [review]
Completely different context and implementation from the original commit, but accomplishes the same thing.

Lgtm.

**Comment by phohensee** [comment]
I tagged the issue with an approval request (added jdk8u-fix-request label), have to wait for a maintainer to approve before integrating. In the future, please wait until maintainer approval before adding "/integrate".

**Comment by yathamravali** [comment]
Sure, I'll make a note of it. 

Tagging maintainers(https://wiki.openjdk.org/display/jdk8u)  @theRealAph @gnu-andrew @jerboaa for their approval.

**Comment by yathamravali** [comment]
@phohensee - Noticed that the JBS issue has the approval tag `jdk8u-fix-yes`. Could you please sponsor if there isn't any pending action?

**Pull request closed without merging**
